src/Stripe.net/Entities/Coupons/Coupon.cs

```diff
@@ -7,56 +7,113 @@ namespace Stripe
 
     public class Coupon : StripeEntity<Coupon>, IHasId, IHasMetadata, IHasObject
     {
+        /// <summary>
+        /// Unique identifier for the object.
+        /// </summary>
         [JsonProperty("id")]
         public string Id { get; set; }
 
+        /// <summary>
+        /// String representing the object's type. Objects of the same type share the same value.
+        /// </summary>
         [JsonProperty("object")]
         public string Object { get; set; }
 
+        /// <summary>
+        /// Amount (in the <c>currency</c> specified) that will be taken off the subtotal of any
+        /// invoices for this customer.
+        /// </summary>
         [JsonProperty("amount_off")]
         public long? AmountOff { get; set; }
 
+        [JsonProperty("applies_to")]
+        public CouponAppliesTo AppliesTo { get; set; }
+
+        /// <summary>
+        /// Time at which the object was created. Measured in seconds since the Unix epoch.
+        /// </summary>
         [JsonProperty("created")]
         [JsonConverter(typeof(DateTimeConverter))]
         public DateTime Created { get; set; }
 
+        /// <summary>
+        /// If <c>amount_off</c> has been set, the three-letter <a
+        /// href="https://stripe.com/docs/currencies">ISO code for the currency</a> of the amount to
+        /// take off.
+        /// </summary>
         [JsonProperty("currency")]
         public string Currency { get; set; }
 
-        /// <summary>
-        /// Whether this object is deleted or not.
-        /// </summary>
         [JsonProperty("deleted", NullValueHandling = NullValueHandling.Ignore)]
```

Comment:
We should do an override for those @cjavilla-stripe as the comments shouldn't disappear